File: single-language-analysis/dataset-files/java/TestSmell155.java
import org.junit.Test;
import static org.junit.Assert.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class TestSmell155 {

    @Test
    public void testAddPost(WebDriver site, String xSelector1, String xSelector2, String btnSelector, String createBtn, String postTitleInput, String postDescriptionInput, String postContentInput, String postSaveBtn, String postTitleSelector) {
        WebElement createPostBtn = site.findElement(By.cssSelector(createBtn));
        createPostBtn.click();
        try {
            Thread.sleep(testdata.get("wait"));
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        WebElement titleInput = site.findElement(By.xpath(postTitleInput));
        titleInput.sendKeys(postTitle);

        WebElement descriptionInput = site.findElement(By.xpath(postDescriptionInput));
        descriptionInput.sendKeys(postDescription);

        WebElement contentInput = site.findElement(By.xpath(postContentInput));
        contentInput.sendKeys(postContent);
        try {
            Thread.sleep(testdata.get("wait"));
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        WebDriverWait wait = new WebDriverWait(site, 10);
        WebElement saveButton = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(postSaveBtn)));
        saveButton.click();

        WebElement savedPostTitle = wait.until(
            ExpectedConditions.presenceOfElementLocated(By.xpath("//h1[text()='" + postTitle + "']"))
        );

        WebElement postTitleElement = site.findElement(By.xpath(postTitleSelector));
        assertEquals(postTitle, postTitleElement.getText());
    }
}

File: single-language-analysis/dataset-files/python/test_smell_155.py
# Original URL: https://github.com/OlgaAlbul/autoTestOnPython/blob/16d833239941c75f0847aa7de88ea1ad53a38aea/hw2/test_2.py#L55-L84

def test_smell_155(site, x_selector1, x_selector2, btn_selector, create_btn, post_title_input, post_description_input, post_content_input, post_save_btn, post_title_selector):
    # Click on the post creation button
    create_post_btn = site.find_element("css", create_btn)
    create_post_btn.click()
    time.sleep(testdata["wait"])

    # Enter post data
    title_input = site.find_element("xpath", post_title_input)
    title_input.send_keys(post_title)

    description_input = site.find_element("xpath", post_description_input)
    description_input.send_keys(post_description)

    content_input = site.find_element("xpath", post_content_input)
    content_input.send_keys(post_content)
    time.sleep(testdata["wait"])

    # Save the post
    wait = WebDriverWait(site.driver, 10)
    save_button = wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, post_save_btn)))
    save_button.click()

    # Wait until the new post title becomes visible to verify that the post was saved
    saved_post_title = wait.until(
        EC.presence_of_element_located((By.XPATH, f"//h1[text()='{post_title}']"))
    )

    # Check that the new post appeared with the given name
    post_title_element = site.find_element("xpath", post_title_selector)
    assert post_title_element.text == post_title